apps/prairielearn/assets/scripts/instructorAssessmentAccessClient.ts

```diff
@@ -0,0 +1,216 @@
+  const editAccessRuleModal = document.querySelector('#editAccessRuleModal');
+  const deleteAccessRuleModal = document.querySelector('.js-delete-access-rule-modal');
+  const addRuleButton = document.getElementById('addRuleButton');
+
+  const accessRulesData = decodeData('access-rules-data');
+
+  const ptHost = (accessRulesTable as HTMLElement)?.dataset.ptHost ?? '';
+  const devMode = (accessRulesTable as HTMLElement)?.dataset.devMode === 'true';
+  const hasCourseInstancePermissionView =
+    (accessRulesTable as HTMLElement)?.dataset.hasCourseInstancePermissionView === 'true';
+  const timezone = (accessRulesTable as HTMLElement)?.dataset.timezone ?? 'UTC';
+
+  let editMode = false;
+
+  function refreshTable() {
+    morphdom(
+      accessRulesTable as Node,
+      AccessRulesTable({
+        accessRules: accessRulesData,
+        ptHost,
+        devMode,
+        hasCourseInstancePermissionView,
+        editMode,
+        timezone,
+      }).toString(),
+    );
+  }
+
+  enableEditButton?.addEventListener('click', () => {
+    editMode = true;
+    enableEditButton.style.display = 'none';
+    editModeButtons?.style.removeProperty('display');
+    addRuleButton?.style.removeProperty('display');
+    refreshTable();
+  });
+
+  on('click', '#saveAndSyncButton', () => {
+    const form = document.getElementById('accessRulesForm') as HTMLFormElement;
+    const assessmentAccessRulesInput = form.querySelector(
+      'input[name="assessment_access_rules"]',
+    ) as HTMLInputElement;
+    if (!assessmentAccessRulesInput) return;
+    const accessRulesMap = accessRulesData.map((accessRule: Record<string, any>) => {
+      const rule = {
+        mode: accessRule.assessment_access_rule.mode,
+        uids: accessRule.assessment_access_rule.uids
+          ? accessRule.assessment_access_rule.uids
+          : null,
+        startDate: adjustedDate(accessRule.assessment_access_rule.start_date)
+          .toISOString()
+          .slice(0, 19),
+        endDate: adjustedDate(accessRule.assessment_access_rule.end_date)
+          .toISOString()
+          .slice(0, 19),
+        active: accessRule.assessment_access_rule.active,
+        credit: parseInt(accessRule.assessment_access_rule.credit),
+        timeLimitMin: parseInt(accessRule.assessment_access_rule.time_limit_mins),
+        password: accessRule.assessment_access_rule.password,
+        examUuid: accessRule.assessment_access_rule.exam_uuid,
+      };
+      const filteredRules = Object.fromEntries(Object.entries(rule).filter(([_, value]) => value));
+      return filteredRules;
+    });
+    assessmentAccessRulesInput.value = JSON.stringify(accessRulesMap);
+    form.submit();
+  });
+
+  function adjustedDate(dateString: string | Date) {
+    const date = new Date(dateString);
+    const timezoneOffset = date.getTimezoneOffset() * 60000;
+    return new Date(date.getTime() - timezoneOffset);
+  }
+
+  function handleUpdateAccessRule({ form }: { form: HTMLFormElement }) {
```

Comment:
For things other than component functions, it's ok to pass individual arguments, especially when there's just one of them:

```suggestion
  function handleUpdateAccessRule(form: HTMLFormElement) {
```

Reply:
Updated to pass the individual argument in [179c438](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/179c4381ed706b6df2c049dfd2fcbc3e0ebcce4f).